src/upload/catalog/controller/extension/payment/webpay_rest.php

```diff
@@ -84,15 +86,32 @@ public function callback() {
         if (($this->request->server['REQUEST_METHOD'] == 'POST')) {
             $tokenWs = isset($this->request->post['token_ws']) ? $this->request->post['token_ws'] : null;
         }
-
+        
+        if (isset($_POST['TBK_ID_SESION'])) {
+            $comment = array(
+                'error' => 'Compra cancelada',
+                'detail' => 'La compra ha sido cancelada por el usuario durante el proceso de pago'
+            );
+    
+            $orderStatusId = $this->config->get('payment_webpay_rest_canceled_order_status');
+            $orderComment = 'Pago cancelado: ' . json_encode($comment);
+            $orderNotifyToUser = false;
+    
+            $this->model_checkout_order->addOrderHistory($orderId, $orderStatusId, $orderComment, $orderNotifyToUser);
+            $this->session->data['reject_text'] = 'Pago cancelado por el usuario';
+            $this->session->data['reject_date'] = date('d-m-Y');
+            $this->session->data['reject_time'] = date('H:i:s');
+            $this->failView();
+            return;
+        }
         if (!isset($tokenWs)) {
 
             $comment = array(
                 'error' => 'Compra cancelada',
                 'detail' => 'El token no ha sido enviado'
             );
 
-            $orderStatusId = $this->config->get('payment_webpay_canceled_order_status');
+            $orderStatusId = $this->config->get('payment_webpay_rest_canceled_order_status');
```

Comment:
I'm not sure we should be changing these values to `webpay_rest`, eg. why `payment_webpay_rest_canceled_order_status` instead of the previous `payment_webpay_canceled_order_status`? do we expect the user to have both SDKs working at the same time? should the user have implemented the SOAP SDK, will this make it more work to migrate from SOAP to REST?

Reply:
We do not 'expect' that from the user, but it can happen depending on how he updates.

Reply:
Is the value of the string different?

Reply:
As this plugin can be installed within the original, it needs a different name for all configurations to prevent conflicts.

Reply:
I mean the value.

Reply:
Yes, the value of that config is not the same for each key.  The config value of 'payment_webpay_canceled_order_status' could have a different value than 'payment_webpay_rest_canceled_order_status' config.

Reply:
I mean the default value.
Is it the same or different? Does the merchant need to have 2 different order status id's, one for purchases made through SOAP and another id for REST ones? Would that be useful information or would it just be annoying when trying to see the statuses your orders are in?

Reply:
The default values are the same. The user then can change one or the other on each plugin. 
if we use the same name, the user will change the value on the plugin A, and the value on the plugin B will change too, and that is not what we want. Each plugin should ahve their own value if the user changes the default value.